GoogleForCreators/web-stories-wp
# Partially add animations to fashion template

## Summary
This PR adds the initial animations to the first two pages of the fashion template, as well as necessary tweaks and updates to achieve them.

## Relevant Technical Choices
Updated animation parts to be more unit agnostic, but have a default unit that gets applied.

Also changed display elements to be `contain: layout;` vs `contain: layout paint` which although not as performant removes mask like behavior from element transforming outside of its original bounds. This should also fix the issue where we were getting culling on typefaces when the line height was below 1. Open to other suggestions here too, but figure with this alteration the small loss in perf is worth the benefit in behavior

## To-do
Pretty much the only animation that's keeping this implementation from being 95% of the mockup is the staggered split line animations. There are several ways we can accomplish this each with their own tradeoffs. Gonna get feedback on what we're prioritizing for v1 and then choose option that gets us there quickest. 

Figure for now tho, the current implementation is a good first step.

## User-facing changes
There should be none here.

## Testing Instructions
first run on your local server and **make sure no animations play** (since the PR should keep the `enableAnimation` flag disabled)

Then, set the `enableAnimation` flag in `includes/Dashboard.php` to true and hover over the fashion template with a mouse or click into the details view to see the animations on the first two pages.

---

Fixes #
https://app.zenhub.com/workspaces/web-stories-5e94d3aced449034e1e9b226/issues/google/web-stories-wp/2213

## Screenshots

![fashion_template-1-2](https://user-images.githubusercontent.com/35983235/84192429-3669c700-aa57-11ea-827b-f52e5d6d53eb.gif)

## Initial Mockups
https://drive.google.com/drive/folders/1E0nLATcplhGNzgteOymx3s2SyA-cc7_M
